models/karma.py

```diff
@@ -1,49 +1,58 @@
+# pyright: reportImportCycles=false
+# we need to import User for type checking but that gives a circular import
+
+from datetime import datetime
+
 from pytz import timezone, utc
-from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, func
+from sqlalchemy import ForeignKey, func
 from sqlalchemy.ext.hybrid import hybrid_property
-from sqlalchemy.orm import relationship
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from typing_extensions import TYPE_CHECKING
```

Comment:
Use `typing .TYPE_CHECKING` instead? typing_extensions isn't in the requirements list anyway iirc